### PR TITLE
feat(search): Client-level search method

### PR DESCRIPTION
**Pull request by VladislavFitz**
**Summary**

Adds client-level search method which allows to perform multiple hits and facet searches in a single network request

### CODE REVIEW

File: Sources/AlgoliaSearchClient/Operation/BrowseSynonyms.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
+//
+//  BrowseSynonyms.swift
+//
+//  Created by Vladislav Fitc on 02/08/2021.
+//
+
+import Foundation
+
+class BrowseSynonymsOperation: AsyncOperation {
```
[the remark this answers is not in the document]
Comment (VladislavFitz) [reply Sources/AlgoliaSearchClient/Operation/BrowseSynonyms.swift]: It doesn't. Removed it.

File: Tests/AlgoliaSearchClientTests/Unit/Transport/AlgoliaRetryStrategyTests.swift [hunk 2]
```diff
@@ -134,39 +134,39 @@ class AlgoliaRetryStrategyTests: XCTestCase {
...
+//          operation.completionBlock = {
+//            allOperationsFinishedExpectation.fulfill()
+//          }
+//          operationQueue.addOperation(operation)
+//        }
+//      }
+//    }
+//
+//    waitForExpectations(timeout: 20, handler: nil)
+//  }
```
[the remark this answers is not in the document]
Comment (VladislavFitz) [reply Tests/AlgoliaSearchClientTests/Unit/Transport/AlgoliaRetryStrategyTests.swift]: Done ✅

File: Sources/AlgoliaSearchClient/Models/Search/MultipleIndex/IndexedQuery.swift [hunk 3]
```diff
@@ -9,12 +9,37 @@ import Foundation
 
 public struct IndexedQuery {
 
+  public enum QueryType {
```
[the remark this answers is not in the document]
Comment (VladislavFitz) [reply Sources/AlgoliaSearchClient/Models/Search/MultipleIndex/IndexedQuery.swift]: Good idea. I Introduced `IndexedFacetQuery` and `MultiSearchQuery` types which wraps either an `IndexedQuery` or and `IndexedFacetQuery`. WDYT?

[the remark this answers is not in the document]
Comment (Aallam) [reply Sources/AlgoliaSearchClient/Models/Search/MultipleIndex/IndexedQuery.swift]: Yes, this offers better DX! I like it 👍

**Pull request merged**